libs/libc/modlib/modlib_load.c

```diff
@@ -271,16 +293,29 @@ int modlib_load(FAR struct mod_loadinfo_s *loadinfo)
         }
     }
 #else
-  loadinfo->textalloc = (uintptr_t)lib_malloc(loadinfo->textsize +
-                                              loadinfo->datasize);
+  align = loadinfo->textalign;
+  if (align < loadinfo->dataalign)
+    {
+      align = loadinfo->dataalign;
+    }
+
+  text_size = loadinfo->textsize;
+  if (loadinfo->datasize > 0)
+    {
+      text_size = _ALIGN_UP(text_size, loadinfo->dataalign);
+    }
+
+  loadinfo->textalloc = (uintptr_t)lib_memalign(align,
```

Comment:
Is it good to combine text and data in one allocation? Since it require more large consecutive memory.

Reply:
i dunno. i guess it has both of pros and cons.

iirc, the original code was doing a single allocation and i changed it for CONFIG_ARCH_USE_MODULE_TEXT case.

Reply:
i'm ok with making it always use separate allocations. do you want me to do it in this PR? i'm feeling that a separate PR is more appropriate.

Reply:
Ok, it's fine to put into a new PR.